test/utils/update/beneficial_owners_managing_officers_data_fetch.spec.ts

```diff
@@ -1,39 +1,115 @@
+    mockLoggerError.mockReset();
     mockGetCompanyPscService.mockReset();
     mockGetCompanyOfficers.mockReset();
+    mockGetBeneficialOwnersPrivateData.mockReset();
   });
 
   test("retrieveBoAndMoData sets BO & MO data in appData.update, sets appData.update.bo_mo_data_fetched", async () => {
     appData = {};
     mockGetCompanyPscService.mockReturnValue(MOCK_GET_COMPANY_PSC_ALL_BO_TYPES);
     mockGetCompanyOfficers.mockReturnValue(MOCK_GET_COMPANY_OFFICERS);
+    mockGetBeneficialOwnersPrivateData.mockReturnValue(MOCK_GET_COMPANY_PSC_ALL_BO_TYPES);
     await retrieveBoAndMoData(req, appData);
     expect(appData.update?.review_beneficial_owners_individual?.length).toEqual(1);
     expect(appData.update?.review_beneficial_owners_corporate?.length).toEqual(1);
     expect(appData.update?.review_beneficial_owners_government_or_public_authority?.length).toEqual(1);
     expect(appData.update?.review_managing_officers_individual?.length).toEqual(1);
     expect(appData.update?.review_managing_officers_corporate?.length).toEqual(1);
+    expect(appData.update?.review_managing_officers_corporate?.length).toEqual(1);
+    const usual_residential_address = appData.update?.review_beneficial_owners_individual?.[0].usual_residential_address;
+    expect(usual_residential_address?.property_name_number).toEqual(undefined);
+    expect(appData.update?.bo_mo_data_fetched).toBe(true);
+  });
+
+  test("retrieveBoAndMoData sets BO private data in appData", async () => {
+    appData = { "transaction_id": "123", "overseas_entity_id": "456" };
+    mockGetCompanyPscService.mockReturnValue(MOCK_GET_COMPANY_PSC_ALL_BO_TYPES);
+    mockGetCompanyOfficers.mockReturnValue(MOCK_GET_COMPANY_OFFICERS);
+    mockGetBeneficialOwnersPrivateData.mockReturnValue(PRIVATE_BO_MOCK_DATA);
+    await retrieveBoAndMoData(req, appData);
+    expect(appData.update?.review_beneficial_owners_individual?.length).toEqual(1);
+    const usual_residential_address = appData.update?.review_beneficial_owners_individual?.[0].usual_residential_address;
+    expect(usual_residential_address?.property_name_number).toEqual("REAGAN HICKMAN");
+    expect(usual_residential_address?.line_1).toEqual("72 COWLEY AVENUE");
+    expect(usual_residential_address?.line_2).toEqual("QUIA EX ESSE SINT EU");
+    expect(usual_residential_address?.town).toEqual("AD EUM DEBITIS EST E");
+    expect(usual_residential_address?.country).toEqual("KUWAIT");
+    expect(usual_residential_address?.county).toEqual("ULLAM DOLORUM CUPIDA");
+    expect(usual_residential_address?.postcode).toEqual("76022");
     expect(appData.update?.bo_mo_data_fetched).toBe(true);
   });
 
+  test("Should not set BO data in appData if no Company PSCs returned", async () => {
+    appData = { "transaction_id": "123", "overseas_entity_id": "456" };
+    mockGetCompanyPscService.mockReturnValue({ "items": [] });
+    await retrieveBoAndMoData(req, appData);
+
+    await retrieveBeneficialOwners(req, appData);
```

Comment:
Which function are we testing here?

Reply:
will take out line 73, psc check is in retrieveBeneficialOwners, thanks!